module2/src/com/epam/decomposition/LogicBuilder.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.decomposition;
+
+import com.epam.sorting.ArraySorter;
+
+public class LogicBuilder {
+
+    //task 1
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber) {
+        return ((firstNumber * secondNumber) / (SupportingLogicalBuilder.
+                findGreatestCommonDivisor(firstNumber, secondNumber)));
+    }
+
+    //task 2
+    public static int findLeastCommonMultipleForFourNumbers(int firstNumber, int secondNumber,
+                                                             int thirdNumber, int fourthNumber) {
+        return SupportingLogicalBuilder.findGreatestCommonDivisor((SupportingLogicalBuilder.
+                findGreatestCommonDivisor(firstNumber, secondNumber)),
+                (SupportingLogicalBuilder.findGreatestCommonDivisor(thirdNumber, fourthNumber)));
+    }
+
+    //task 3
+    public static double findHexagonArea(double sizeHexagon, double amountOfCorners) {
+        return SupportingLogicalBuilder.calculateEquilateralTriangleArea(sizeHexagon) * amountOfCorners;
+    }
```

Comment:
a hexagon may not have 6 corners? Why is the amountOfCorners parameter here?
+ you don't really need this SupportingLogicalBuilder and even additional method as you have the solution like 1 line.